src/tests.rs

```diff
@@ -17,6 +17,14 @@ use rand_core::OsRng;
 use std::collections::HashSet;
 use proptest::prelude::*;
 
+// Configure proptest to run fewer cases for faster testing
+// Default is 256 cases, we'll use 8 for quicker feedback  
+// You can also set PROPTEST_CASES=32 environment variable
+// to increase the number of cases for a particular run
+fn fast_config() -> ProptestConfig {
+    ProptestConfig::with_cases(8)
```

Comment:
With this change, the tests finish in ~40 seconds on my laptop (M3 macbook), which is much more reasonable.

Based on my understanding of what these tests are doing (running code on random test cases), in my opinion, you could get away with setting this to just one case per run. You end up running the code against a lot of random test cases over time, as you run `cargo test` more and more. Besides, running 8 test cases instead of 1 means your 8 times as likely to come across a failure, but given the kind of code we're testing, I'm not sure this 8-fold increase makes a huge difference.